src/commands/init.js

```diff
@@ -41,56 +99,17 @@ exports.builder = (yargs) => {
     })
 }
 
+exports.handlerTasks = handlerTasks;
+
 exports.handler = function ({ reporter, name, template }) {
   name = defaultAPMName(name)
   
   const basename = name.split('.')[0]
   const tasks = new TaskList([
-    {
-      title: 'Checking project existence',
-      task: async (ctx, task) => {
-        const projectPath = path.resolve(process.cwd(), basename)
-        const exists = await fs.pathExists(projectPath)
-        if (exists) {
-          throw new Error(`Couldn't initialize project. Project with name ${basename} already exists in ${projectPath}. Use different <name> or rename existing project folder.`)
-        }
-      }
-    },
-    {
-      title: 'Clone template',
-      task: async (ctx, task) => {
-        task.output = `Cloning ${template} into ${basename}...`
-
-        const repo = await clone(template, basename, { shallow: true })
-      }
-    },
-    {
-      title: 'Preparing template',
-      task: async (ctx, task) => {
-        // Set `appName` in arapp
-        const arappPath = path.resolve(
-          basename,
-          'arapp.json'
-        )
-        const arapp = await fs.readJson(arappPath)
-        arapp.appName = name
-
-        // Delete .git folder
-        const gitFolderPath = path.resolve(
-          basename,
-          '.git'
-        )
-
-        return Promise.all([
-          fs.writeJson(arappPath, arapp, { spaces: 2 }),
-          fs.remove(gitFolderPath)
-        ])
-      }
-    },
-    {
-      title: 'Install package dependencies',
-      task: async (ctx, task) => (await installDeps(basename, task)),
-    }
+    handlerTasks.checkProjectExistence({ basename }),
```

Comment:
Just a note: in some commands, context from one task to another is passed by writing to the `ctx` object. In order to test individual tasks, the context will need to be properly set. Example: https://github.com/aragon/aragon-cli/blob/master/src/commands/apm_cmds/publish.js#L217

Reply:
👍